Enterprise-CMCS/managed-care-review
# MCR-4152, MCR-4231 Add form,  LinkRateSelect, and action buttons to replace rate

## Summary
This is MCR-4231 and an initial pass of work for MCR-4152. Add on PR will be needed to complete AC.

### Acceptance criteria covered:
#### MCR-4152
1. ✅ If I am an admin user on the submission summary page 
Then I see a "replace rate" button next to each rate review
2.  ✅ If I click on "replace rate" 
Then I am navigated to a screen where I see an form field to "Add a rate review to replace this rate" where I can select from a dropdown all the rate reviews for that state
3. ❌  If I am viewing the dropdown of rate reviews to replace the rate with
Then only rates for the the state whose submission I am on should appear in the dropdown, not rates from other states 
4.  ✅ If I type in part of the rate name into the typeahead 
Then the dropdown options narrow to the criteria of my search 
5.  ✅ If I make a selection and click "continue" 
Then I am navigated back to the submission summary page and the new rate I selected appears on the submission summary page.  
6. ❌ We call the new API successfully
7.  ✅ The withdrawn rate is not shown on submission summary or unlocked submissions (for either CMS or State users)

#### MCR-4231
 ✅ If a rate has been withdrawn for in a state 
Then if I am logged in as a user in that state and I start a new submission then I do not see the withdrawn rate appear on the linked rate dropdown on the rate details page

#### Related issues
https://jiraent.cms.gov/browse/MCR-4152
https://jiraent.cms.gov/browse/MCR-4231

#### Screenshots
- Don't bother reviewing for styles - can add polish and adjust margin later if we decide its priority. Here is current state:
<img width="736" alt="Screenshot 2024-08-08 at 12 00 11 PM" src="https://github.com/user-attachments/assets/2a2984c2-547a-4c67-9da1-253ffdcee2aa">

#### Test cases covered
- Not React component tests yet
- I honed in some of the API tests a bit, I have plans to add more if I verify multi-rates is broken 

## QA guidance
- Please review for the AC with the green check mark only
- Please help me ensure this PR does not cause regressions for linked rates --  verify in whatever way makes sense. I did touch the component used for the connecting linked rates
